code/src/pgsql_io/pgsql_io.ml

```diff
@@ -991,7 +1163,17 @@ and create_sm_process_login_frames ?passwd ~user t =
   | AuthenticationSCMCredential :: _ ->
       Abb.Future.return (Error `Unsupported_auth_scm_credential_err)
   | AuthenticationGSS :: _ -> Abb.Future.return (Error `Unsupported_auth_gss_err)
-  | AuthenticationSASL _ :: _ -> Abb.Future.return (Error `Unsupported_auth_sasl_err)
+  | AuthenticationSASL { auth_mechanisms = [ ("SCRAM-SHA-256" as auth_mechanism) ] } :: fs -> (
```

Comment:
Because the `PLUS` variant might be there, this should probably be:

`| AuthenticationSASL { auth_mechanisms  } :: fs when CCList.mem ~eq:CCString.equal "SCRAM-SHA-256" auth_mechanisms -> ...`